etcdmain/etcd.go

```diff
@@ -210,20 +210,17 @@ func startEtcd(cfg *config) (<-chan struct{}, error) {
 		if u.Scheme == "http" && !cfg.peerTLSInfo.Empty() {
 			plog.Warningf("The scheme of peer url %s is http while peer key/cert files are presented. Ignored peer key/cert files.", u.String())
 		}
+
+		urlStr := u.String()
+		plog.Info("listening for peers on ", urlStr)
+
 		var l net.Listener
 		l, err = rafthttp.NewListener(u, cfg.peerTLSInfo)
 		if err != nil {
+			l.Close()
+			plog.Info("stopping listening for peers on ", urlStr)
```

Comment:
How about just check err for log writing and remove defer()? defer in a for loop is al little confusing.